src/main/java/jenkins/scm/api/SCMHeadEvent.java

```diff
@@ -170,47 +169,27 @@ public static void fireNow(@NonNull final SCMHeadEvent<?> event) {
      * @param delayUnits the units of time in which the delay is expressed.
      */
     public static void fireLater(@NonNull final SCMHeadEvent<?> event, long delay, TimeUnit delayUnits) {
-        executorService().schedule(new Dispatcher(event), delay, delayUnits);
+        executorService().schedule(new DispatcherImpl(event), delay, delayUnits);
     }
 
-    private static class Dispatcher implements Runnable {
-        private final SCMHeadEvent<?> event;
+    private static class DispatcherImpl extends Dispatcher<SCMHeadEvent<?>> {
+        private DispatcherImpl(SCMHeadEvent<?> event) {
+            super(event);
+        }
 
-        public Dispatcher(SCMHeadEvent<?> event) {
-            this.event = event;
+        @Override
+        protected void log(SCMEventListener l, Throwable e) {
+            LogRecord lr = new LogRecord(Level.WARNING,
+                    "SCMEventListener.onSCMHeadEvent(SCMHeadEvent) {0} propagated an exception"
+            );
+            lr.setThrown(e);
+            lr.setParameters(new Object[]{l});
+            LOGGER.log(lr);
```

Comment:
Simpler to write:

```java
LOGGER.log(Level.WARNING, "SCMEventListener.onSCMHeadEvent(SCMHeadEvent) " + l + " propagated an exception", e);
```

Since `INFO` and above are logged by default, there is no expected performance savings from this more verbose idiom.